src/dumbo/src/tcp/handler.rs

```diff
@@ -493,6 +516,16 @@ mod tests {
         }
     }
 
+    fn mock_callback(request: Result<Request, RequestError>) -> Response {
```

Comment:
If we want to create a mock callback, wouldn't it be enough to return some `Response` type (for example `Response::new(Version::Http11, StatusCode::OK)`) and not even check the `request` variant via that match?

Also, this function looks like a duplicate. Can we get rid of one of them?